webapp/channels/src/components/admin_console/team_channel_settings/convert_and_remove_confirm_modal.tsx

```diff
@@ -43,34 +41,48 @@ type Props = {
 };
 
 const ConvertAndRemoveConfirmModal = ({show, onConfirm, onCancel, displayName, toPublic, removeAmount}: Props) => {
-    const toPublicMsg = 'When you convert **{displayName}** to a public channel, history and membership are preserved. Public channels are discoverable and can by joined by users on the system without invitation.  \n \nAre you sure you want to convert **{displayName}** to a public channel?';
-    const toPrivateMsg = 'When you convert **{displayName}** to a private channel, history and membership are preserved. Publicly shared files remain accessible to anyone with the link. Membership in a private channel is by invitation only.  \n \nAre you sure you want to convert **{displayName}** to a private channel?';
-    const convertMessageId = toPublic ? t('admin.team_channel_settings.convertConfirmModal.toPublicMessage') : t('admin.team_channel_settings.convertConfirmModal.toPrivateMessage');
-
-    const toPublicTitle = 'Convert channel to public and remove {amount, number} {amount, plural, one {user} other {users}}?';
-    const toPrivateTitle = 'Convert channel to private and remove {amount, number} {amount, plural, one {user} other {users}}?';
-    const titleId = toPublic ? t('admin.team_channel_settings.convertAndRemoveConfirmModal.toPublicTitle') : t('admin.team_channel_settings.convertAndRemoveConfirmModal.toPrivateTitle');
-
-    const toPublicConfirmMsg = 'Yes, convert channel to public and remove {amount, number} {amount, plural, one {user} other {users}}';
-    const toPrivateConfirmMsg = 'Yes, convert channel to private and remove {amount, number} {amount, plural, one {user} other {users}}';
-    const confirmMsgId = toPublic ? t('admin.team_channel_settings.convertAndRemoveConfirmModal.toPublicConfirm') : t('admin.team_channel_settings.convertAndRemoveConfirmModal.toPrivateConfirm');
-
-    const title = (
-        <FormattedMessage
-            id={titleId}
-            defaultMessage={toPublic ? toPublicTitle : toPrivateTitle}
-            values={{displayName, amount: removeAmount}}
-        />
-    );
+    let title;
+    if (toPublic) {
+        title = (
+            <FormattedMessage
+                id='admin.team_channel_settings.convertAndRemoveConfirmModal.toPublicTitle'
+                defaultMessage='Convert channel to public and remove {amount, number} {amount, plural, one {user} other {users}}?'
+                values={{amount: removeAmount}}
+            />
+        );
+    } else {
+        title = (
+            <FormattedMessage
+                id='admin.team_channel_settings.convertAndRemoveConfirmModal.toPrivateTitle'
+                defaultMessage='Convert channel to private and remove {amount, number} {amount, plural, one {user} other {users}}?'
+                values={{amount: removeAmount}}
+            />
+        );
+    }
+
+    let convertMessage;
+    if (toPublic) {
+        convertMessage = (
+            <FormattedMarkdownMessage
```

Comment:
There's still some things here that would be worth fixing such as removing `FormattedMarkdownMessage` and fixing some weird whitespace in this string that the FormatJS ESLint plugin will complain about, but this diff is difficult enough to read as-is, so I'm not changing it yet